From 8e044f2e92fc4a768602fbfaf5d38f71b71839cd Mon Sep 17 00:00:00 2001
From: James Hartley <james.hartley@imgtec.com>
Date: Wed, 1 Jun 2016 11:24:47 +0100
Subject: FROMLIST: MIPS:pistachio:Remove plat_setup_iocoherency

The Pistachio SoC does not have an IOCU.
Hence, DMA is non-coherent

Remove the function checking for iocoherency and select
CONFIG_DMA_NONCOHERENT in Kconfig

This code is probably inhereted from Malta.

Change-Id: I571c97ef5f0fe3056a50083ccc9ecf95cf0ba465
Signed-off-by: Zubair Lutfullah Kakakhel <Zubair.Kakakhel@imgtec.com>
Signed-off-by: James Hartley <james.hartley@imgtec.com>
---
 arch/mips/Kconfig          |  2 +-
 arch/mips/pistachio/init.c | 25 -------------------------
 2 files changed, 1 insertion(+), 26 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index cd3b14d..1c58731 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -384,7 +384,7 @@ config MACH_PISTACHIO
 	select CLKSRC_MIPS_GIC
 	select COMMON_CLK
 	select CSRC_R4K
-	select DMA_MAYBE_COHERENT
+	select DMA_NONCOHERENT
 	select IRQ_MIPS_CPU
 	select LIBFDT
 	select MFD_SYSCON
diff --git a/arch/mips/pistachio/init.c b/arch/mips/pistachio/init.c
index d41c614..5aefb06 100644
--- a/arch/mips/pistachio/init.c
+++ b/arch/mips/pistachio/init.c
@@ -61,37 +61,12 @@ const char *get_system_type(void)
 	return sys_type;
 }
 
-static void __init plat_setup_iocoherency(void)
-{
-	/*
-	 * Kernel has been configured with software coherency
-	 * but we might choose to turn it off and use hardware
-	 * coherency instead.
-	 */
-	if (mips_cm_numiocu() != 0) {
-		/* Nothing special needs to be done to enable coherency */
-		pr_info("CMP IOCU detected\n");
-		hw_coherentio = 1;
-		if (coherentio == 0)
-			pr_info("Hardware DMA cache coherency disabled\n");
-		else
-			pr_info("Hardware DMA cache coherency enabled\n");
-	} else {
-		if (coherentio == 1)
-			pr_info("Hardware DMA cache coherency unsupported, but enabled from command line!\n");
-		else
-			pr_info("Software DMA cache coherency enabled\n");
-	}
-}
-
 void __init plat_mem_setup(void)
 {
 	if (fw_arg0 != -2)
 		panic("Device-tree not present");
 
 	__dt_setup_arch((void *)fw_arg1);
-
-	plat_setup_iocoherency();
 }
 
 #define DEFAULT_CPC_BASE_ADDR	0x1bde0000
-- 
2.6.2

